code/modules/holodeck/items.dm

```diff
@@ -32,6 +32,22 @@
 /obj/item/melee/energy/sword/holographic/red
 	sword_color_icon = "red"
 
+/obj/item/toy/cards/deck/syndicate/holographic
+	desc = "A deck of holographic playing cards."
+
+/obj/item/toy/cards/deck/syndicate/holographic/Initialize(mapload, obj/machinery/computer/holodeck/holodeck)
+	src.holodeck = holodeck
+	RegisterSignal(src, COMSIG_PARENT_QDELETING, .proc/handle_card_delete)
+	. = ..()
+
+/obj/item/toy/cards/deck/syndicate/holographic/proc/handle_card_delete(datum/source)
+	SIGNAL_HANDLER
+
+	//if any REAL cards have been inserted into the deck they are moved outside before destroying it
+	for(var/obj/item/toy/singlecard/card in cards)
+		cards -= card
+		card.forceMove(drop_location())
```

Comment:
you should only move cards that dont have flags_1 & HOLOGRAM_1 assuming that cards spawned inside a hologram deck start off with that flag. if they dont then they should

Reply:
actually: currently the holodeck doesnt correctly do what i described but it should. you'd need to 1. change /obj/machinery/computer/holodeck/proc/finish_spawn() so that it adds lists returns by holo_effect.activate() to the spawned list, and make sure all objects returned by spawned have HOLOGRAM_1 added to them. then 2. make the deck holoeffects return a list of all of their cards in activate(). then you wont need to manually pass the holodeck to the cards/decks

Reply:
Applied with https://github.com/tgstation/tgstation/pull/64200/commits/045cafcc926c1fb5f7561b3197035157a930085d